### ISSUE TITLE
Affirmative system deps declartion

### ISSUE THREAD

**Issue by whitecolor** [stealjs/steal#393]
Instead of using `npmIngnore` there is need in some affirmative option for example `npmDependencies` that will tell steal only load explicitly stated packages there.

**Comment by matthewp** [comment]
Our plan is to load them progressively so it only loads the npm packages after you `require()` them.  The exception would be your own dependencies so something like this could be useful.

**Comment by justinbmeyer** [comment]
I think something like this could be useful short term too.  In some apps, the number of client packages is much less than the total number of listed packages.  This would be actually less work to specify them.

Ideally, it would be smart enough to work with npmIgnore like:

```
npmIgnore: ["devDependencies"],
npmDependencies: ["steal-qunit"]
```

This would still load `steal-qunit`, and all `dependencies`.  However, the following:

```
// npmIgnore: undefined,
npmDependencies: ["steal-qunit"]
```

Would only load steal-qunit.

Basically, only npmDependencies get loaded if npmIgnore is not used.  If npmIgnore has `dependencies`, `devDependencies`, or `peerDependencies`, `npmDependencies` acts as an override.

**Comment by whitecolor** [comment]
I've have to put in `npmIgnore` 30-40 deps currenlty - it is not nice.

**Comment by matthewp** [comment]
Ok, maybe we can get this in `0.9.0` then, will have to see if it's a small change.

**Comment by matthewp** [comment]
I'm a little leary of this as it seems like it could break the plugin's ability to properly crawl. Consider this scenario in the root package:

### root package.json
```json
{
  "name": "rootPackage",
  "dependencies": {
    "one": "1.0.0",
    "two": "1.0.0"
  },
  "system": {
    "npmDependencies": [ "one" ]
  }
}
```

## one package.json
```json
{
  "name": "one",
  "dependencies": {
    "two": "1.0.0"
  }
}
```

In this scenario both the root package and `one` depend on `two`.  This means two will be inside the root package.json's `node_modules` folder.  However because we did not include `two` in the root package's `npmDependencies` array, when `one` attempts to load it, it will incorrectly look in `node_modules/one/node_modules/two`.

Probably there is a solution to this.  I think we should be looking at parent packages to see if they contain the same semver dependency.

**Comment by whitecolor** [comment]
I belive that if module "two" is need "one" should be parsed. In practice such situation will  be met very rarely, if at all. I belive the same problem with `npmIgnore` no?

**Comment by matthewp** [comment]
Yes, I believe we might be able to fix this scenario and it probably would also happen with `npmIgnore`, i'll test it out and see.

**Comment by matthewp** [comment]
https://github.com/stealjs/system-npm/pull/55

Leaving this open to track documentation being added.

**Comment by matthewp** [comment]
Fixed by #401, will be coming in 0.9.0.
